clients/pkg/promtail/targets/kafka/target.go

```diff
@@ -34,17 +38,21 @@ type Target struct {
 	client               api.EntryHandler
 	relabelConfig        []*relabel.Config
 	useIncomingTimestamp bool
+	messageParser        MessageParser
 }
 
 func NewTarget(
+	logger log.Logger,
 	session sarama.ConsumerGroupSession,
 	claim sarama.ConsumerGroupClaim,
 	discoveredLabels, lbs model.LabelSet,
 	relabelConfig []*relabel.Config,
 	client api.EntryHandler,
 	useIncomingTimestamp bool,
+	messageParser MessageParser,
```

Comment:
for injecting a dependency, I think is always better to use an interface rather than a function type, just to cover cases in which one needs to use something other than a function. For example, you could do:
```go
type MessageParser interface {
  Parse(...)
}
```
and in the azure case just make the function type `func(message *sarama.ConsumerMessage) ([]string, error)` implement that interface

Reply:
Valid point 👍. Agree, it gives more flexibility. 

I changed it and introduced a noopParser for a Kafka target, which just returns the raw message payload as `[]string{string(message.Value)}`